core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -489,8 +486,9 @@ public GetResult get(Get get, Function<String, Searcher> searcherFactory) throws
                         op.id(),
                         op.versionType().explainConflictForWrites(currentVersion, expectedVersion, deleted));
                 result = onFailure.apply(e);
+            } else {
+                result = onSuccess.get();
```

Comment:
can we add a comment here ? something like "version conflicts during recovery and replica operation are normal and occur due to out of order delivery. we should return a successful result"